sdk/identity/Azure.Identity/src/EnvironmentCredential.cs

```diff
@@ -59,13 +59,19 @@ internal EnvironmentCredential(CredentialPipeline pipeline)
             string username = EnvironmentVariables.Username;
             string password = EnvironmentVariables.Password;
             string sdkAuthLocation = EnvironmentVariables.SdkAuthLocation;
+            string clientCertificatePath = EnvironmentVariables.ClientCertificatePath;
+            var clientCertificate = new X509Certificate2(clientCertificatePath);
```

Comment:
The `X509Certificate2` should only be instantiated in the if block below.

~~~ c#                
               else if (clientCertificate != null)
                {
                    var clientCertificate = new X509Certificate2(clientCertificatePath);

                    _credential = new ClientCertificateCredential(tenantId, clientId, clientCertificate, _pipeline);
                }
~~~